File: app/src/main/java/me/tagavari/airmessage/helper/StringHelper.java
package me.tagavari.airmessage.helper;

public class StringHelper {
	public static String nullifyEmptyString(String string) {
		return defaultEmptyString(string, null);
	}
	
	public static String defaultEmptyString(String string, String fallback) {
		if(string != null && string.isEmpty()) return fallback;
		else return string;
	}
	
	public static boolean stringContainsOnlyEmoji(String string) {
		//Ignoring if the string is empty
		if(string.isEmpty()) return false;
		
		//Returning if there are any non-emoji characters in the string
		int length = string.length();
		for(int offset = 0; offset < length;) {
			int codePoint = string.codePointAt(offset);
			if(!isCharEmoji(codePoint) && !isZeroWidthJoiner(codePoint)) return false;
			offset += Character.charCount(codePoint);
		}
		
		//Returning true
		return true;
	}
	
	public static boolean isCharEmoji(int codePoint) {
		return (codePoint >= 0x1F600 && codePoint <= 0x1F64F) ||
				(codePoint >= 0x1F300 && codePoint <= 0x1F5FF) ||
				(codePoint >= 0x1F680 && codePoint <= 0x1F6FF) ||
				(codePoint >= 0x2600 && codePoint <= 0x26FF) ||
				(codePoint >= 0x2700 && codePoint <= 0x27BF) ||
				(codePoint >= 0xE0020 && codePoint <= 0xE007F) ||
				(codePoint >= 0xFE00 && codePoint <= 0xFE0F) ||
				(codePoint >= 0x1F900 && codePoint <= 0x1F9FF) ||
				(codePoint >= 0x1F018 && codePoint <= 0x1F270) ||
				(codePoint >= 0x238C && codePoint <= 0x2454) ||
				(codePoint >= 0x20D0 && codePoint <= 0x20FF);
	}
	
	public static boolean isZeroWidthJoiner(int codePoint) {
		return codePoint == 8205;
	}
}

File: app/src/main/java/me/tagavari/airmessage/helper/StringHelper.kt
package me.tagavari.airmessage.helper

object StringHelper {
	@JvmStatic
	fun nullifyEmptyString(string: String?): String? {
		return if(string.isNullOrEmpty()) null else string
	}
	
	@JvmStatic
	fun defaultEmptyString(string: String?, fallback: String): String {
		return if(string.isNullOrEmpty()) fallback else string
	}
	
	@JvmStatic
	fun stringContainsOnlyEmoji(string: String): Boolean {
		//Ignoring if the string is empty
		if(string.isEmpty()) return false
		
		//Returning if there are any non-emoji characters in the string
		val length = string.length
		var offset = 0
		while(offset < length) {
			val codePoint = string.codePointAt(offset)
			if(!isCharEmoji(codePoint) && !isZeroWidthJoiner(codePoint)) return false
			offset += Character.charCount(codePoint)
		}
		
		//Returning true
		return true
	}
	
	@JvmStatic
	fun isCharEmoji(codePoint: Int) =
			codePoint in 0x1F600..0x1F64F ||
			codePoint in 0x1F300..0x1F5FF ||
			codePoint in 0x1F680..0x1F6FF ||
			codePoint in 0x2600..0x26FF ||
			codePoint in 0x2700..0x27BF ||
			codePoint in 0xE0020..0xE007F ||
			codePoint in 0xFE00..0xFE0F ||
			codePoint in 0x1F900..0x1F9FF ||
			codePoint in 0x1F018..0x1F270 ||
			codePoint in 0x238C..0x2454 ||
			codePoint in 0x20D0..0x20FF
	
	@JvmStatic
	fun isZeroWidthJoiner(codePoint: Int) = codePoint == 8205
}